README.md

```diff
@@ -42,7 +42,7 @@ Default function accepts several options.
 | Name                      | Type          | Default           | Description                                          |
 |---------------------------|---------------|-------------------|------------------------------------------------------|
 | `resolver`                | Function      | `httpsResolver`   | Function used to test if request is secure           |
-| `hostname`                | String        | `undefined`       | Hostname for redirect (uses request host if not set) |
+| `hostname`                | String \| Function        | `undefined`       | Hostname for redirect, or a function that takes `ctx` as its only argument and returns the desired hostname to use. Uses request host if not set or resolved value is falsy |
```

Comment:
I also mentioned it under the issue but I think it might even make sense to retire the old string api. In a way it makes it only harder to understand that there is more flexible option.

Reply:
Seems good to me to go "function only" that if it's good for you.

Do you still like the function signature, accepting a `ctx` as the parameter? You'd suggested the `hostname` in your comment.

Quick thoughts: `ctx` seems like it provides more flexibility, while `hostname` is more restrictive...but that may be good since it restricts people being able to manipulate the `ctx` as a side-effect of `sslify` calling their function.

I don't, personally, feel strongly either way and/or am unsure which is better.

Reply:
> I would maybe pass only the host to the function and expect only the new host or null/undefined back. Just to make it a bit easier to understand and harder to mess things up. If someone needs access to whole `ctx` then I think implementing another middleware would be a better option - that way the control flow (order of middlewares) would be more explicit.

And there's my answer. If you've seen the way I did it (with `ctx`) and still want `hostname`, that's good by me. Will change.

Reply:
I'm open to suggestions from your side as well. You probably had more time to think about it than I did. But yeah this is what I would be in favor of doing at this point.

Reply:
I think `hostname` is fine and agree that if they need more complexity than that, they should/could write their own middleware as that'd be kinda outside the targeted scope of this library.